runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -1175,3 +1177,235 @@ J9::ValuePropagation::getParmValues()
 
    TR_ASSERT(parmIterator->atEnd() && parmIndex == numParms, "Bad signature for owning method");
    }
+
+static bool isTakenSideOfAVirtualGuard(TR::Compilation* comp, TR::Block* block)
+   {
+   // First block can never be taken side
+   if (block == comp->getMethodSymbol()->getFirstTreeTop()->getEnclosingBlock())
+      return false;
+
+   // Due to VirtualGuardHeadMerger, the taken side of a virtual guard may have more than one predecessors,
+   // each containing a virtual guard that branches to the taken side. It's sufficient to look at only
+   // one predecessor.
+   TR::Node* predLastRealNode = block->getPredecessors().size() > 0 ?
```

Comment:
VGHM or other opts may also chain the cold side of guards together - see VirtualGuardTailSplitter - it is not sufficient to look at only one.